### ISSUE TITLE
postgresql - don't use the k8s secret that contains the tls certificates	

### ISSUE THREAD

**Issue by yaakov-berkovitch** [bitnami/charts#3315]
**Which chart**:
postgresql-9.1.2

**Describe the bug**
Install the postgresql chart with a configmap for the postgresql.conf and a kubernetes secret for the tls certificates.
The install failed with the error (from the pod log):
`2020-08-04 02:05:18 EDT  1  FATAL:  could not load server certificate file "server.crt": No such file or directory
`
Seems that the tls certificates are not taken from the k8s secret.

**To Reproduce**
1. Create a values.yml as follows:
```
## Postgres image tag
image:
  tag: 12.3.0-debian-10-r18

postgresqlUsername: postgres

# ConfigMap with the PostgreSQL configuration
configurationConfigMap: postgresql.conf

## PostgreSQL password using existing secret
existingSecret: postgres-secret

# Secret that contains the certificates
tls:
    enabled: true
    certificatesSecret: postgres-certificates-tls
    certFilename: cert.crt
    certKeyFilename: cert.key

## Mount PostgreSQL secret as a file instead of passing environment variable
usePasswordFile: false

persistence:
  enabled: true
  existingClaim: postgres-data-pvc
  storageClass: "postgres-data-pv"

volumePermissions:
  enabled: true

master:
  podAnnotations:
    sidecar.istio.io/inject: "false"

```
2. run the helm install :
`/usr/local/bin//helm install postgresql bitnami/postgresql --version 9.1.2 -f /tmp/postgres-values.yml`
3. Check the statefulset yaml is correctly defined - see the volumes section and the raw-certificates:
```
apiVersion: apps/v1
kind: StatefulSet
metadata:
  annotations:
    meta.helm.sh/release-name: postgresql
    meta.helm.sh/release-namespace: default
  creationTimestamp: "2020-08-04T06:04:21Z"
  generation: 1
  labels:
    app.kubernetes.io/instance: postgresql
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: postgresql
    helm.sh/chart: postgresql-9.1.2
  managedFields:
  - apiVersion: apps/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:meta.helm.sh/release-name: {}
          f:meta.helm.sh/release-namespace: {}
        f:labels:
          .: {}
          f:app.kubernetes.io/instance: {}
          f:app.kubernetes.io/managed-by: {}
          f:app.kubernetes.io/name: {}
          f:helm.sh/chart: {}
      f:spec:
        f:podManagementPolicy: {}
        f:replicas: {}
        f:revisionHistoryLimit: {}
        f:selector:
          f:matchLabels:
            .: {}
            f:app.kubernetes.io/instance: {}
            f:app.kubernetes.io/name: {}
            f:role: {}
        f:serviceName: {}
        f:template:
          f:metadata:
            f:annotations:
              .: {}
              f:sidecar.istio.io/inject: {}
            f:labels:
              .: {}
              f:app.kubernetes.io/instance: {}
              f:app.kubernetes.io/managed-by: {}
              f:app.kubernetes.io/name: {}
              f:helm.sh/chart: {}
              f:role: {}
            f:name: {}
          f:spec:
            f:containers:
              k:{"name":"postgresql"}:
                .: {}
                f:env:
                  .: {}
                  k:{"name":"BITNAMI_DEBUG"}:
                    .: {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"PGDATA"}:
                    .: {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"POSTGRES_PASSWORD"}:
                    .: {}
                    f:name: {}
                    f:valueFrom:
                      .: {}
                      f:secretKeyRef:
                        .: {}
                        f:key: {}
                        f:name: {}
                  k:{"name":"POSTGRES_USER"}:
                    .: {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"POSTGRESQL_ENABLE_LDAP"}:
                    .: {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"POSTGRESQL_ENABLE_TLS"}:
                    .: {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"POSTGRESQL_PORT_NUMBER"}:
                    .: {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"POSTGRESQL_TLS_CERT_FILE"}:
                    .: {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"POSTGRESQL_TLS_KEY_FILE"}:
                    .: {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"POSTGRESQL_TLS_PREFER_SERVER_CIPHERS"}:
                    .: {}
                    f:name: {}
                    f:value: {}
                  k:{"name":"POSTGRESQL_VOLUME_DIR"}:
                    .: {}
                    f:name: {}
                    f:value: {}
                f:image: {}
                f:imagePullPolicy: {}
                f:livenessProbe:
                  .: {}
                  f:exec:
                    .: {}
                    f:command: {}
                  f:failureThreshold: {}
                  f:initialDelaySeconds: {}
                  f:periodSeconds: {}
                  f:successThreshold: {}
                  f:timeoutSeconds: {}
                f:name: {}
                f:ports:
                  .: {}
                  k:{"containerPort":5432,"protocol":"TCP"}:
                    .: {}
                    f:containerPort: {}
                    f:name: {}
                    f:protocol: {}
                f:readinessProbe:
                  .: {}
                  f:exec:
                    .: {}
                    f:command: {}
                  f:failureThreshold: {}
                  f:initialDelaySeconds: {}
                  f:periodSeconds: {}
                  f:successThreshold: {}
                  f:timeoutSeconds: {}
                f:resources:
                  .: {}
                  f:requests:
                    .: {}
                    f:cpu: {}
                    f:memory: {}
                f:securityContext:
                  .: {}
                  f:runAsUser: {}
                f:terminationMessagePath: {}
                f:terminationMessagePolicy: {}
                f:volumeMounts:
                  .: {}
                  k:{"mountPath":"/bitnami/postgresql"}:
                    .: {}
                    f:mountPath: {}
                    f:name: {}
                  k:{"mountPath":"/bitnami/postgresql/conf"}:
                    .: {}
                    f:mountPath: {}
                    f:name: {}
                  k:{"mountPath":"/dev/shm"}:
                    .: {}
                    f:mountPath: {}
                    f:name: {}
                  k:{"mountPath":"/opt/bitnami/postgresql/certs"}:
                    .: {}
                    f:mountPath: {}
                    f:name: {}
                    f:readOnly: {}
            f:dnsPolicy: {}
            f:initContainers:
              .: {}
              k:{"name":"init-chmod-data"}:
                .: {}
                f:command: {}
                f:image: {}
                f:imagePullPolicy: {}
                f:name: {}
                f:resources:
                  .: {}
                  f:requests:
                    .: {}
                    f:cpu: {}
                    f:memory: {}
                f:securityContext:
                  .: {}
                  f:runAsUser: {}
                f:terminationMessagePath: {}
                f:terminationMessagePolicy: {}
                f:volumeMounts:
                  .: {}
                  k:{"mountPath":"/bitnami/postgresql"}:
                    .: {}
                    f:mountPath: {}
                    f:name: {}
                  k:{"mountPath":"/dev/shm"}:
                    .: {}
                    f:mountPath: {}
                    f:name: {}
                  k:{"mountPath":"/opt/bitnami/postgresql/certs"}:
                    .: {}
                    f:mountPath: {}
                    f:name: {}
                  k:{"mountPath":"/tmp/certs"}:
                    .: {}
                    f:mountPath: {}
                    f:name: {}
            f:restartPolicy: {}
            f:schedulerName: {}
            f:securityContext:
              .: {}
              f:fsGroup: {}
            f:terminationGracePeriodSeconds: {}
            f:volumes:
              .: {}
              k:{"name":"data"}:
                .: {}
                f:name: {}
                f:persistentVolumeClaim:
                  .: {}
                  f:claimName: {}
              k:{"name":"dshm"}:
                .: {}
                f:emptyDir:
                  .: {}
                  f:medium: {}
                  f:sizeLimit: {}
                f:name: {}
              k:{"name":"postgresql-certificates"}:
                .: {}
                f:emptyDir: {}
                f:name: {}
              k:{"name":"postgresql-config"}:
                .: {}
                f:configMap:
                  .: {}
                  f:defaultMode: {}
                  f:name: {}
                f:name: {}
              k:{"name":"raw-certificates"}:
                .: {}
                f:name: {}
                f:secret:
                  .: {}
                  f:defaultMode: {}
                  f:secretName: {}
        f:updateStrategy:
          f:type: {}
    manager: Go-http-client
    operation: Update
    time: "2020-08-04T06:04:21Z"
  - apiVersion: apps/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:status:
        f:collisionCount: {}
        f:currentReplicas: {}
        f:currentRevision: {}
        f:observedGeneration: {}
        f:replicas: {}
        f:updateRevision: {}
        f:updatedReplicas: {}
    manager: kube-controller-manager
    operation: Update
    time: "2020-08-04T06:04:21Z"
  name: postgresql-postgresql
  namespace: default
  resourceVersion: "170915"
  selfLink: /apis/apps/v1/namespaces/default/statefulsets/postgresql-postgresql
  uid: 74110d48-f096-4a50-8c16-95fa63a35eed
spec:
  podManagementPolicy: OrderedReady
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app.kubernetes.io/instance: postgresql
      app.kubernetes.io/name: postgresql
      role: master
  serviceName: postgresql-headless
  template:
    metadata:
      annotations:
        sidecar.istio.io/inject: "false"
      creationTimestamp: null
      labels:
        app.kubernetes.io/instance: postgresql
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: postgresql
        helm.sh/chart: postgresql-9.1.2
        role: master
      name: postgresql
    spec:
      containers:
      - env:
        - name: BITNAMI_DEBUG
          value: "false"
        - name: POSTGRESQL_PORT_NUMBER
          value: "5432"
        - name: POSTGRESQL_VOLUME_DIR
          value: /bitnami/postgresql
        - name: PGDATA
          value: /bitnami/postgresql/data
        - name: POSTGRES_USER
          value: postgres
        - name: POSTGRES_PASSWORD
          valueFrom:
            secretKeyRef:
              key: postgresql-password
              name: postgres-secret
        - name: POSTGRESQL_ENABLE_LDAP
          value: "no"
        - name: POSTGRESQL_ENABLE_TLS
          value: "yes"
        - name: POSTGRESQL_TLS_PREFER_SERVER_CIPHERS
          value: "yes"
        - name: POSTGRESQL_TLS_CERT_FILE
          value: /opt/bitnami/postgresql/certs/cert.crt
        - name: POSTGRESQL_TLS_KEY_FILE
          value: /opt/bitnami/postgresql/certs/cert.key
        image: docker.io/bitnami/postgresql:12.3.0-debian-10-r18
        imagePullPolicy: IfNotPresent
        livenessProbe:
          exec:
            command:
            - /bin/sh
            - -c
            - exec pg_isready -U "postgres" -h 127.0.0.1 -p 5432
          failureThreshold: 6
          initialDelaySeconds: 30
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 5
        name: postgresql
        ports:
        - containerPort: 5432
          name: tcp-postgresql
          protocol: TCP
        readinessProbe:
          exec:
            command:
            - /bin/sh
            - -c
            - -e
            - |
              exec pg_isready -U "postgres" -h 127.0.0.1 -p 5432
              [ -f /opt/bitnami/postgresql/tmp/.initialized ] || [ -f /bitnami/postgresql/.initialized ]
          failureThreshold: 6
          initialDelaySeconds: 5
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 5
        resources:
          requests:
            cpu: 250m
            memory: 256Mi
        securityContext:
          runAsUser: 1001
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /opt/bitnami/postgresql/certs
          name: postgresql-certificates
          readOnly: true
        - mountPath: /dev/shm
          name: dshm
        - mountPath: /bitnami/postgresql
          name: data
        - mountPath: /bitnami/postgresql/conf
          name: postgresql-config
      dnsPolicy: ClusterFirst
      initContainers:
      - command:
        - /bin/sh
        - -cx
        - |
          mkdir -p /bitnami/postgresql/data /bitnami/postgresql/conf
          chmod 700 /bitnami/postgresql/data /bitnami/postgresql/conf
          find /bitnami/postgresql -mindepth 1 -maxdepth 1 -not -name ".snapshot" -not -name "lost+found" | \
            xargs chown -R 1001:1001
          chmod -R 777 /dev/shm
          cp /tmp/certs/* /opt/bitnami/postgresql/certs/
          chown -R 1001:1001 /opt/bitnami/postgresql/certs/
          chmod 600 /opt/bitnami/postgresql/certs/cert.key
        image: docker.io/bitnami/minideb:buster
        imagePullPolicy: Always
        name: init-chmod-data
        resources:
          requests:
            cpu: 250m
            memory: 256Mi
        securityContext:
          runAsUser: 0
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /bitnami/postgresql
          name: data
        - mountPath: /dev/shm
          name: dshm
        - mountPath: /tmp/certs
          name: raw-certificates
        - mountPath: /opt/bitnami/postgresql/certs
          name: postgresql-certificates
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext:
        fsGroup: 1001
      terminationGracePeriodSeconds: 30
      volumes:
      - configMap:
          defaultMode: 420
          name: postgresql.conf
        name: postgresql-config
      - name: raw-certificates
        secret:
          defaultMode: 420
          secretName: postgres-certificates-tls
      - emptyDir: {}
        name: postgresql-certificates
      - emptyDir:
          medium: Memory
          sizeLimit: 1Gi
        name: dshm
      - name: data
        persistentVolumeClaim:
          claimName: postgres-data-pvc
  updateStrategy:
    type: RollingUpdate
status:
  collisionCount: 0
  currentReplicas: 1
  currentRevision: postgresql-postgresql-654879576c
  observedGeneration: 1
  replicas: 1
  updateRevision: postgresql-postgresql-654879576c
  updatedReplicas: 1

```
4. Pod logs:
```
postgresql 06:15:20.09
postgresql 06:15:20.09 Welcome to the Bitnami postgresql container
postgresql 06:15:20.10 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql
postgresql 06:15:20.10 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql/issues
postgresql 06:15:20.10
postgresql 06:15:20.11 INFO  ==> ** Starting PostgreSQL setup **
postgresql 06:15:20.18 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql 06:15:20.18 INFO  ==> Loading custom pre-init scripts...
postgresql 06:15:20.19 INFO  ==> Initializing PostgreSQL database...
postgresql 06:15:20.20 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
postgresql 06:15:20.22 INFO  ==> pg_hba.conf file not detected. Generating it...
postgresql 06:15:20.22 INFO  ==> Generating local authentication configuration
postgresql 06:15:20.23 INFO  ==> Deploying PostgreSQL with persisted data...
postgresql 06:15:20.24 INFO  ==> Loading custom scripts...
postgresql 06:15:20.24 INFO  ==> Stopping PostgreSQL...
postgresql 06:15:20.25 INFO  ==> ** PostgreSQL setup finished! **

postgresql 06:15:20.31 INFO  ==> ** Starting PostgreSQL **
2020-08-04 02:15:20 EDT  1  FATAL:  could not load server certificate file "server.crt": No such file or directory
2020-08-04 02:15:20 EDT  1  LOG:  database system is shut down

````
**Expected behavior**
postgresql installed and tls sertificates read from the k8s secret and not from a file in the data folder.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.2.3", GitCommit:"8f832046e258e2cb800894579b1b3b50c2d83492", GitTreeState:"clean", GoVersion:"go1.13.12"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.4", GitCommit:"c96aede7b5205121079932896c4ad89bb93260af", GitTreeState:"clean", BuildDate:"2020-06-17T11:41:22Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:51:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by marcosbc** [comment]
Hi @yaakov-berkovitch, it looks like you are specifying a really old tag of the PostgreSQL docker image. [This feature was implemented in `12.3.0-debian-10-r53`](https://github.com/bitnami/bitnami-docker-postgresql/commit/10a9002f8b2ed885a24b2ec8feb0f79f0b0a8dbf) but you are using `12.3.0-debian-10-r18`.

Also, note that the feature is not going to work when you mount `postgresql.conf` because it will not be able to modify it, so it is skipped.

Therefore, in order to fix the issue on your side, you will need to manually update your `postgresql.conf` configmap to reflect the proper value of the SSL/TLS certificate configurations, since the container scripts will refuse to do so.

**Comment by yaakov-berkovitch** [comment]
I had a doubt if the tls certificates configuration can be overridden while also using postgresql.conf. It was not clear for me while reading the documentation.
Thanks for your hints.
